pkg/controller/machineset/mapper.go

```diff
@@ -69,3 +69,8 @@ func (m *msmapper) hostMatchesMachineSet(host *bmh.BareMetalHost, ms *machinev1b
 	}
 	return selector.Matches(labels.Set(host.ObjectMeta.Labels)), nil
 }
+
+var mapperFunc = func(obj client.Object) []reconcile.Request {
+	mapper := msmapper{}
```

Comment:
This has to be initialised with a client in order to do anything.

Reply:
Fixed this.